gio/gappinfo.c

```diff
@@ -824,12 +846,68 @@ launch_default_with_portal_async (const char          *uri,
   g_free (real_uri);
 }
 
+static void
+launch_default_with_portal_sync (const char          *uri,
+                                 GAppLaunchContext   *context)
+{
+  GDBusConnection *session_bus;
+  GVariantBuilder opt_builder;
+  GVariant *res = NULL;
+  const char *parent_window = NULL;
+  char *real_uri;
+  GError *error = NULL;
+
+  session_bus = g_bus_get_sync (G_BUS_TYPE_SESSION, NULL, &error);
+  if (session_bus == NULL)
+    {
+      g_task_report_error (context, NULL, NULL, NULL, error);
+      return;
+    }
+
+  if (context && context->priv->envp)
+    parent_window = g_environ_getenv (context->priv->envp, "PARENT_WINDOW_ID");
+
+  real_uri = real_uri_for_portal (uri, context, NULL, NULL, NULL, &error);
+  if (real_uri == NULL)
+    return;
```

Comment:
Good catch. This line is just a refactor of the original code so that's why I did not see it, but it's true is a mistake, [and one that is present on the original code](https://github.com/endlessm/glib/blob/master/gio/gappinfo.c#L804). I'll fix it in a separate commit, previous to this one, to make the fix explicit.

About the g_autoptr, we can't use it here, see my previous comment.